lib/aws-manager.js

```diff
@@ -81,142 +82,164 @@ class AwsManager {
+            {
+              Name: 'instance-state-name',
+              Values: ['running', 'pending'],
+            },
+          ],
+        }).promise(),
+        // Living spot requests
+        obj.describeSpotInstanceRequests({
+          Filters: [
+            {
+              Name: 'launch.key-name',
+              Values: [this.keyPrefix + '*'],
+            }, {
+              Name: 'state',
+              Values: ['open'],
+            },
+          ],
+        }).promise(),
+        // Dead instances
+        obj.describeInstances({
+          Filters: [
+            {
+              Name: 'key-name',
+              Values: [this.keyPrefix + '*'],
+            },
+            {
+              Name: 'instance-state-name',
+              Values: ['shutting-down', 'terminated', 'stopping'],
+            },
+          ],
+        }).promise(),
+        // Dead spot requests
+        obj.describeSpotInstaceRequests({
+          Filters: [
+            {
+              Name: 'launch.key-name',
+              Values: [this.keyPrefix + '*'],
+            }, {
+              Name: 'state',
+              Values: ['cancelled', 'failed', 'closed', 'active'],
+            },
+          ],
+        }).promise(),
+        // Available availability zones
+        obj.describeAvailabilityZones({
+          Filters: [
+            {
+              Name: 'state',
+              Values: ['available'],
+            },
+          ],
+        }).promise(),
+        // Raw pricing data
+        obj.describeSpotPriceHistory({
+          StartTime: pricingStartDate,
+          Filters: [
+            {
+              Name: 'product-description',
+              Values: ['Linux/UNIX'],
+            },
+          ],
+        }).promise(),
+      ]);
 
-      deadInstances = await this.ec2.describeInstances({
-        Filters: [
-          {
-            Name: 'key-name',
-            Values: [this.keyPrefix + '*'],
-          },
-          {
-            Name: 'instance-state-name',
-            Values: ['shutting-down', 'terminated', 'stopping'],
-          },
-        ],
-      });
-      debug('fetched dead instance state');
+      // Now let's classify them
 
-      let rawDeadSR = await this.ec2.describeSpotInstanceRequests({
-        Filters: [
-          {
-            Name: 'launch.key-name',
-            Values: [this.keyPrefix + '*'],
-          }, {
-            Name: 'state',
-            Values: ['cancelled', 'failed', 'closed', 'active'],
-          },
-        ],
-      });
-      debug('fetched dead spot request state');
+      for (let reservation of response[0].data.Reservations) {
+        for (let instance of reservation.Instances) {
+          let workerType = this.parseKeyPairName(instance.KeyName).workerType;
+          // Maybe use objFilter here
+          let filtered = instance;
+          filtered.Region = region;
+          filtered.WorkerType = workerType;
+          apiState.instances.push(filtered);
+        }
+      };
```

Comment:
Rather than `apiState.instances.push(filtered)` let's see if we can't do `.map` instead..

```js
apiState.instances = _.flatten(_.map(response[0].data.Reservations, _.property('Instances'))).map(instance => {
  instance.WorkerType = this.parseKeyPairName(instance.KeyName).workerType;
  instance.Region = region;
  return instance;
});
```
If we use the `.then(_.property('data.Reservations'))` proposed earlier this becomes: 
```js
apiState.instances = _.flatten(_.map(livingResevations, _.property('Instances'))).map(inst => {
  inst.WorkerType = this.parseKeyPairName(instance.KeyName).workerType;
  inst.Region = region;
  return inst;
});
```

I propose this because it's now very clear that `apiState.instances` is set here... and you don't get the idea that it's partially populated before this...